src/lakefs_spec/spec.py

```diff
@@ -332,6 +476,27 @@ def put_file_to_blockstore(
         presign: bool = False,
         storage_options: dict[str, Any] | None = None,
     ) -> None:
+        """
+        Upload a file to lakeFS by directly putting it into its underlying block storage, thereby reducing the request load
+        on the lakeFS server.
+
+        Requires the corresponding fsspec implementation for the block storage type used by your lakeFS server deployment.
+
+        Supported block storage types are S3 (needs ``s3fs``), GCS (needs ``gcsfs``), and Azure Blob Storage (needs ``adlfs``).
+
+        Parameters
+        ----------
+        lpath: str
+            The local path to upload to the lakeFS block storage.
+        rpath: str
+            The remote target path to upload the local file to. Must be a fully qualified lakeFS URI.
+        callback: fsspec.callbacks.Callback
+            An fsspec callback to use during the operation. Can be used to report download progress.
+        presign: bool
+            Whether to use pre-signed URLs to upload the object via HTTP(S) using ``urllib.request``.
```

Comment:
Mention the implications if `presign=False`: must have locally available credentials for accessing the blockstore (right?)